src/internal/modules/waiting/controller.js

```diff
@@ -1,71 +1,95 @@
+  return title + ' bill run';
+};
+
+const getWaitingForBilling = async (request, h, event) => {
+  const path = getRedirectPath({
+    'complete': '/billing/batch/summary'
+  }, event);
+  if (path) {
+    return h.redirect(path);
+  }
+  const { data } = await services.water.billingBatchCreateService.getBillingRegions();
+
   const view = {
     ...request.view,
-    pageTitle: getPageTitle(event),
-    text: 'Please wait while the mailing list is assembled. This may take a few minutes. The letters will not be sent yet.'
+    pageTitle: getBillingTitle(event, data),
+    caption: moment(event.date_created).format('D MMM YYYY'),
+    waitingType: 'billRun'
   };
+
   return h.view('nunjucks/waiting/index', view);
 };
 
-const handleReturnReminderError = (request, h) => {
-  throw Boom.badImplementation('Errored event.');
+const getWaitingForNotifications = async (request, h, event) => {
+  const path = getRedirectPath({
+    'processed': '/batch-notifications/review'
+  }, event);
+  if (path) {
+    return h.redirect(path);
+  }
+
+  const view = {
+    ...request.view,
+    pageTitle: getNotificationsTitle(event),
+    text: 'Please wait while the mailing list is assembled. This may take a few minutes. The letters will not be sent yet.',
+    waitingType: 'returns'
+  };
+
+  return h.view('nunjucks/waiting/index', view);
 };
 
-const handleReturnsRemindersProcessed = (request, h, event) => {
-  // Redirect to a new page showing a send button and a means
-  // of acquiring a csv download of all the recipients.
-  const { event_id: eventId } = event;
-  return h.redirect(`/batch-notifications/review/${eventId}`);
+const getNotificationsTitle = (ev) => {
+  const name = get(ev, 'subtype');
+  const config = {
+    returnReminder: 'Send returns reminders',
+    returnInvitation: 'Send returns invitations'
+  };
+  return config[name];
 };
 
-const subTypeHandlers = {
-  returnReminder: {
-    processing: handleProcessing,
-    processed: handleReturnsRemindersProcessed,
-    error: handleReturnReminderError
-  },
-  returnInvitation: {
-    processing: handleProcessing,
-    processed: handleReturnsRemindersProcessed,
-    error: handleReturnReminderError
-  }
+const getRegionName = (regionsArray, regionId) => {
+  const [ region ] = regionsArray.filter(region => region.regionId === regionId);
+  return region.name;
 };
 
-const getEventHandler = event => {
-  return get(subTypeHandlers, [event.subtype, event.status]);
+const handlers = {
+  notification: getWaitingForNotifications,
+  'billing-batch': getWaitingForBilling
 };
 
 const getWaiting = async (request, h) => {
   const { eventId } = request.params;
   const { data: event, error } = await services.water.events.findOne(eventId);
 
-  throwIfError(error);
-
-  const handler = getEventHandler(event);
-
-  if (!handler) {
+  if (error) {
     const message = 'Unknown event type';
     logger.error(message, { params: event });
```

Comment:
`logger.error` should be called with message, error, params so here this should be:

`logger.error(message, error, event);`

Reply:
done